src/utils/bytes.h

```diff
@@ -0,0 +1,10 @@
+#ifndef BDK_UTILS_BYTES_H
+#define BDK_UTILS_BYTES_H
+
+#include <vector>
+
+using Byte = uint8_t;
+
+using Bytes = std::vector<Byte>;
+
```

Comment:
Friendly suggestion:

```suggestion
template <std::size_t N> using BytesArr = std::array<Byte, N>; ///< Typedef for BytesArr.
```

This is gonna help me a bit on my `optimize` branch, been duplicating those lines everywhere